addons/l10n_in/models/account_invoice.py

```diff
@@ -67,6 +80,18 @@ def _compute_l10n_in_state_id(self):
             else:
                 move.l10n_in_state_id = False
 
+    @api.depends('partner_id', 'partner_shipping_id', 'company_id', 'l10n_in_state_id')
+    def _compute_fiscal_position_id(self):
+        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.journal_id.type == 'sale')
+        for move in in_moves:
+            pos_partner = move._get_l10n_in_pos_sale_partner()
+            move.fiscal_position_id = self.env['account.fiscal.position'].with_company(move.company_id
+            ).with_context(
+                force_state=move.l10n_in_state_id,
+                force_country=pos_partner.country_id or move.company_id.country_id
+            )._get_fiscal_position(pos_partner)
```

Comment:
Why aren't we just computing all that in `_get_fiscal_position` instead of playing with the context?